### PR TITLE
Feature/cloud 1389 sonar project and scan

### PR DESCRIPTION
**Pull request by naveen-variant**
# Description

Code changes to create sonar project in sonarcloud if it is not available.

Fixes [#24](https://usxtech.atlassian.net/browse/CLOUD-1389)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

```yaml
Test Case 1 [Test on dotnet projects]:
- Steps
  - Delete sonar project in sonar cloud ui.
  - Go to any demo app and create a branch to the actions-collection branch or use the branch i created.
  - Update the workflow file to point to actions-dotnet branch of enable sonar.
  - Run the workflow. 
- Result
 - See project is created in sonarcloud console with the run results pushed.
 - sample run https://github.com/variant-inc/demo-app/actions/runs/1730483138
 
Test Case 2 [Test on nodejs projects]:
- Steps
  - Go to any demo app and create a branch to the actions-collection branch or use the branch i created.
  - Update the workflow file to point to actions-nodejs branch of enable sonar.
  - Run the workflow. 
- Result
 - See project is created in sonarcloud console with the run results pushed.
 - Sample run https://github.com/variant-inc/demo-nodejs-app/actions/runs/1730788560

Test Case 3 [Test on python projects]:
- Steps
  - Go to any demo app and create a branch to the actions-collection branch or use the branch i created.
  - Update the workflow file to point to actions-python branch of enable sonar.
  - Run the workflow. 
- Result
 - See project is created in sonarcloud console with the run results pushed.
 - sample run https://github.com/variant-inc/demo-python-flask-variant-api/actions/runs/1741915207
```

- [x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added documentation to test

### CODE REVIEW

File: scripts/enable_sonar.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+echo "Running provision_projects"
+curl --fail --include \
+    --request POST \
+    --header "Content-Type: application/x-www-form-urlencoded" \
+    -u ${SONAR_TOKEN}: \
+    --data-binary "installationKeys=${GITHUB_REPOSITORY}%7C${REPO_ID}&organization=${SONAR_ORG}" \
+    "https://sonarcloud.io/api/alm_integration/provision_projects"
+
+SONAR_PROJECT_KEY=$(echo ${GITHUB_REPOSITORY} | sed 's/\//_/')
```
Comment (vibindaniel) [inline scripts/enable_sonar.sh:24]: Where do you get this from https://github.com/variant-inc/actions-dotnet/blob/f6c038b536e1895d4ae03990a57220c7a4b3fe92/scripts/sonarscan.sh#L16?

Comment (naveen-variant) [reply scripts/enable_sonar.sh:24]: We get sonar project key from here https://github.com/variant-inc/actions-setup/blob/8a0f48d7e99ea62109d16a35440980af9c2f2370/action.yml#L20. Basically i am doing same thing here while creating sonar project.

**Comment by salyateem13** [comment]
@naveen-variant Ran last demo-app run on master branch and it did not recreate the sonar project. Any ideas?

File: scripts/enable_sonar.sh (hunk 1 above)
Comment (rchengvariant) [reply scripts/enable_sonar.sh]: Is this removed?

**Comment by salyateem13** [comment]
Do we need all sonar projects prepended with variant-inc? This will be a little annoying if you manage a lot of projects

**Review by salyateem13 — APPROVED** [review]

**Review by rchengvariant — APPROVED** [review]

**Comment by naveen-variant** [comment]
> Do we need all sonar projects prepended with variant-inc? This will be a little annoying if you manage a lot of projects

Its the format of github repo name we get from git env variable

**Pull request merged**